crates/utils/re_video/src/demux/mod.rs

```diff
@@ -509,6 +507,43 @@ impl VideoDataDescription {
         self.samples.num_elements()
     }
 
+    /// Duration of the video.
+    ///
+    /// For videos that may still grow this is an estimate of the video length so far.
+    /// (we don't know for sure how long the last sample is going to be)
+    pub fn duration(&self) -> Option<std::time::Duration> {
+        match &self.update_type {
+            VideoUpdateType::NoUpdates { duration } => Some(duration.duration(self.timescale?)),
+
+            VideoUpdateType::Stream { .. } => {
+                if self.samples.num_elements() < 2 {
+                    return None;
```

Comment:
A zero-frame video should have duration `std::time::Duration::ZERO`.

A single-frame video should return `SampleMetadata::duration`

Reply:
which would mean that we alternative between some and none just because a frame was added.
Let's have it be `None` only if the timescale is `None` and `Some` in _all_ other cases